products/ocp4/profiles/pci-dss-node.profile

```diff
@@ -0,0 +1,21 @@
+documentation_complete: true
+
+platform: ocp4
+
+metadata:
+    SMEs:
+        - JAORMX
+        - jhrozek
+        - mrogers950
+
+reference: https://www.pcisecuritystandards.org/documents/PCI_DSS_v3-2-1.pdf
+
+title: 'PCI-DSS v3.2.1 Control Baseline for Red Hat OpenShift Container Platform 4'
+
+description: |-
+    Ensures PCI-DSS v3.2.1 security configuration settings are applied.
+
+filter_rules: 'platform == "ocp4-node"'
```

Comment:
It feels strange to have this separated from the control selection item (i.e. `pcidss_ocp4:all:base`).

Do we also want/need the `filter_rules` to work independently of control based profiles?

Reply:
Reading the description again there is no mention of the controls file. I read the PR with a different context in mind, :stuck_out_tongue:

Reply:
We have discussed the question of interface - should this filtering clause be here, in the profile file, or should it be closer to the controls manipulation, because that's what we want - to influence what rules we get from those controls?

I suggest that we keep it as it is for now - we can easily change the interface to something else, as the code that asks rules about their platform would stay, and only the place it is invoked from would change.
By looking at `pcidss_ocp4:all:base`, it is clear that the control syntax has been abused enough, and `pcidss_ocp4:all:base:platform == something` could be the solution for now, but wouldn't benefit the project in the long-term. So let's come with a better-looking and more robust and intuitive instantiation syntax, and only then let's use it.

Reply:
> We have discussed the question of interface - should this filtering clause be here, in the profile file, or should it be closer to the controls manipulation, because that's what we want - to influence what rules we get from those controls?

TL;DR; I tend to think that that the filtering clause should be in the profile.
--
The control file keeps a set of controls organized according to the policy it is tracking, it organizes the controls into policy levels and maps rules to them. It can be seen as an organized "rule gatherer".

Now the pool of rules gathered by the control file need to make it into profiles.
And, although the profiles can be aligned to the policy (ANSSI), they can be organized in an unrelated way (PCI-DSS for OCP4). What I mean by that is that the separation into `master` and `node` is an OCP4 construct, the policy doesn't predict the product being "divided" into two parts.

So I think the filter should be defined in the profile to avoid adding product specific constructs in the control file.

> By looking at `pcidss_ocp4:all:base`, it is clear that the control syntax has been abused enough, and `pcidss_ocp4:all:base:platform == something` could be the solution for now, but wouldn't benefit the project in the long-term - that would be the third selection option added to the one-line syntax, as first we select `all` controls, we then restrict those with `level == base`, and then we have this platform filter. It is likely that another selection requirements are yet to be discovered. So let's come with a better-looking and more robust and intuitive instantiation syntax, and only then let's use it.

I agree that the control syntax is being pushed beyond what it was designed for, which, as far as I remember, was to be backward compatible with the profile syntax.
Maybe consider backward compatibility breaking changes?